Upstream-Status: Pending [Not submitted to upstream yet]
Signed-off-by: Abdellatif El Khlifi <abdellatif.elkhlifi@arm.com>

From 32969f7963890fc0e3c7d47d788751272d95e501 Mon Sep 17 00:00:00 2001
From: Rui Miguel Silva <rui.silva@linaro.org>
Date: Fri, 20 Dec 2019 13:12:27 +0000
Subject: [PATCH 17/20] corstone700_defconfig: enable configs for cramfs-xip

Add the MTD and CRAMFS support for CramFS-XIP

Because the NOR flash is already mapped and read-accessible
before the kernel starts, so we don't need a QSPI driver to
access the rootfs. The kernel shouldn't be configured with
MTD_CFI. Otherwise, the kernel will verify CFI compliancy
by writing commands to the flash and a QSPI driver is required
in this case.

Signed-off-by: Abdellatif El Khlifi <abdellatif.elkhlifi@arm.com>
Signed-off-by: Rui Miguel Silva <rui.silva@linaro.org>
---
 arch/arm/configs/corstone700_defconfig | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/arch/arm/configs/corstone700_defconfig b/arch/arm/configs/corstone700_defconfig
index add9f5673395..13863b804c6b 100644
--- a/arch/arm/configs/corstone700_defconfig
+++ b/arch/arm/configs/corstone700_defconfig
@@ -60,13 +60,17 @@ CONFIG_IP_PNP_BOOTP=y
 # CONFIG_NET_9P_VIRTIO=y
 # CONFIG_UEVENT_HELPER_PATH="/sbin/hotplug"
 CONFIG_DEVTMPFS=y
-# CONFIG_MTD=y
+CONFIG_MTD=y
 # CONFIG_MTD_CMDLINE_PARTS=y
 # CONFIG_MTD_BLOCK=y
-# CONFIG_MTD_CFI=y
+CONFIG_MTD_RAM=y
+CONFIG_MTD_ROM=y
 # CONFIG_MTD_CFI_INTELEXT=y
 # CONFIG_MTD_CFI_AMDSTD=y
-# CONFIG_MTD_PHYSMAP=y
+CONFIG_MTD_PHYSMAP=y
+CONFIG_MTD_PHYSMAP_COMPAT=y
+CONFIG_MTD_PHYSMAP_START=0x08100000
+CONFIG_MTD_PHYSMAP_LEN=0x200000
 # CONFIG_MTD_PHYSMAP_OF=y
 # CONFIG_MTD_PLATRAM=y
 # CONFIG_MTD_UBI=y
@@ -149,7 +153,8 @@ CONFIG_RTC_DRV_PL031=y
 # CONFIG_TMPFS=y
 # CONFIG_JFFS2_FS=y
 # CONFIG_UBIFS_FS=y
-# CONFIG_CRAMFS=y
+CONFIG_CRAMFS=y
+CONFIG_CRAMFS_MTD=y
 # CONFIG_SQUASHFS=y
 # CONFIG_SQUASHFS_LZO=y
 # CONFIG_NFS_FS=y
-- 
2.17.1

